### PR TITLE
Add Support for Pip Packaging + Fused Reorder Kernel Support

### PR DESCRIPTION
**Pull request by elvircrn**
The goals of this PR are:

* Simplify C++ bindings.
* Prepare the repository for pip packaging.
* Resolve an UB in the legacy conversion C++ function.
* Get rid of old setup.py.
* Rename all mentions of buff0 into dense_matrix (makes it more descriptive).
* Delete CMakeLists as it's only needed for local debugging/IDE support.
* Add support for fused reordering.

### CODE REVIEW

File: inference_lib/README.md [hunk 1]
```diff
@@ -0,0 +1,252 @@
+# SpQR model compression
+
+
+**Note:** This repository contains quantization algorithm and the model evaluation code for SpQR method for LLM compression;
```
Comment (Vahe1994) [inline inference_lib/README.md:4]: Why did you copy readme here without changes?

File: inference_lib/README.md [hunk 2]
```diff
@@ -0,0 +1,252 @@
+# SpQR model compression
+
+
+**Note:** This repository contains quantization algorithm and the model evaluation code for SpQR method for LLM compression; 
+The efficient inference code will be added soon.
```
Comment (Vahe1994) [inline inference_lib/README.md:5]: Isn't this readme for inference code ?:)

File: inference_lib/README.md [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+    --qq_scale_bits 3 \
+    --qq_zero_bits 3 \
+    --qq_groupsize 16 \
+    --outlier_threshold=0.2 \
+    --permutation_order act_order \
+    --percdamp 1e0 \
+    --nsamples 128 
+```
+The command above runs near-lossless compression as described in the article. Adjusting the above parameters allows for tighter compression with a slightly greater loss. 
+
```
Comment (Vahe1994) [inline inference_lib/README.md:85]: I don't think you need all of this here

File: inference_lib/setup.cfg [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+[options]
+package_dir =
+    = src
+packages = find:
+include_package_data = True
+python_requires = >=3.11
+install_requires =
+    torch>=2.4.0
+    transformers>=4.44.0
+    accelerate>=0.27.0
```
Comment (Vahe1994) [inline inference_lib/setup.cfg:35]: Do you need accelerate?

Comment (elvircrn) [reply inference_lib/setup.cfg]: No, need for it.

File: inference_lib/README.md (hunk 2 above)
Comment (elvircrn) [reply inference_lib/README.md]: Resolved by reworking the README

**Pull request merged**